core/src/main/java/com/linecorp/armeria/server/HttpResponseSubscriber.java

```diff
@@ -200,6 +207,21 @@ public void onNext(HttpObject o) {
                     newHeaders.remove(HttpHeaderNames.CONTENT_LENGTH);
                 }
 
+                if (!newHeaders.contains(HttpHeaderNames.SERVER) &&
+                    config.defaultServerNameResponseHeader()) {
```

Comment:
Could you check the `config` first, which is relatively cheaper than `headers.contains()`?

Reply:
Fixed it. Thank you for education!